File: two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        
        mapp = dict()
        
        for i in range(len(nums)):
            n = nums[i]
            complement = target - n
            if n in mapp:
                return [mapp[n], i]
            else:
                mapp[complement] = i
            
        #time o(n) space o(n)

File: two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        Map<Integer, Integer> mapp = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++){
            if(mapp.containsKey(target - nums[i])){
                res[1] = i;
                res[0] = mapp.get(target - nums[i]);
                return res;
            }
            else{
                mapp.put(nums[i], i);
            }
        } 
        return res;
    }
}